tests/security/swtpm/swtpm_env_setup.pm

```diff
@@ -0,0 +1,71 @@
+# Copyright (C) 2021 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+#
+# Summary: Ship the "swtpm" software TPM emulator for QEMU,
+#          install required packages and download the pre-installed
+#          OS images for later tests, prepare a script for remote
+#          ssh login and execution
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#81256, tc#1768671
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils 'zypper_call';
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Install qemu, libvirt, swtpm, expect and virsh_tools packages
+    # "net-tools-deprecated" is now back to legacy module in SLE
+    zypper_call("in qemu libvirt swtpm expect virt-install virt-manager");
+
+    # Start libvirtd daemon and start the default libvirt network
+    assert_script_run("systemctl start libvirtd");
+    assert_script_run("virsh net-start default");
+    assert_script_run("systemctl is-active libvirtd");
```

Comment:
Should be before the `virsh` call?

Reply:
Yes, it should be

Reply:
By default, the libvirtd daemon should not be started by default after installation. so I need start it along with the default network "virbr0".

I will remain this code, since I didn't see any function impact there. please free free to reopen and add your suggestion here if you don't agree with me.